src/main/java/io/vertx/core/spi/logging/LogDelegate.java

```diff
@@ -72,4 +72,15 @@
   void trace(Object message, Throwable t);
 
   void trace(Object message, Throwable t, Object... params);
+
+  /**
+   * Give access to a typed instance of the underlying framework logger object.
+   *
+   * @param cls the expected type, or an interface the logger object implements
+   * @return a typed logger instance, null in the default implementation
+   * @throws ClassCastException if the logger object is not assignable to {@code cls}
+   */
+  default <T> T unwrap(Class<T> cls) {
```

Comment:
just return Object, users will do:

```
Foo foo = (Foo) delegate.unwrap();
```

instead of:

```
Foo foo = delegate.unwrap(Foo.class);
```

the later looks too smart for a cast, let just use the cast operator Java has been providing for years.